### PR TITLE
[5.1] Improvements to the request class

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

File: tests/Http/HttpRequestTest.php [hunk 1]
```diff
@@ -315,8 +315,6 @@ public function testJSONEmulatingPHPBuiltInServer()
     {
         $payload = ['name' => 'taylor'];
         $content = json_encode($payload);
-        // The built in PHP 5.4 webserver incorrectly provides HTTP_CONTENT_TYPE and HTTP_CONTENT_LENGTH,
-        // rather than CONTENT_TYPE and CONTENT_LENGTH
```
Comment (franzliedke) [inline tests/Http/HttpRequestTest.php]: Out of interest: why did you remove these, Graham? I find that comments like this help understand why certain strange test cases are added...

Comment (GrahamCampbell) [inline tests/Http/HttpRequestTest.php]: It's because it's not the build in webserver, that's just how php does headers, and that's commonly known now.

Comment (franzliedke) [inline tests/Http/HttpRequestTest.php]: Okay, thanks.
